File: src/main/java/play/pluv/login/domain/TesterLoginValidator.java
package play.pluv.login.domain;

import static org.apache.commons.codec.digest.DigestUtils.sha256Hex;
import static play.pluv.login.exception.LoginExceptionType.INVALID_TESTER_ID_PASSWORD;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import play.pluv.login.exception.LoginException;

@Component
public class TesterLoginValidator {

  private final String validSecret;
  private final Long testerId;

  public TesterLoginValidator(
      @Value("${tester.secret}") final String validSecret,
      @Value("${tester.id}") final Long testerId
  ) {
    this.validSecret = validSecret;
    this.testerId = testerId;
  }

  public Long getTesterId(final String id, final String password) {
    final String validSecret = sha256Hex(id + password);
    if (!this.validSecret.equals(validSecret)) {
      throw new LoginException(INVALID_TESTER_ID_PASSWORD);
    }
    return testerId;
  }
}

File: src/main/kotlin/play/pluv/login/domain/TesterLoginValidator.kt
package play.pluv.login.domain

import org.apache.commons.codec.digest.DigestUtils
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import play.pluv.login.exception.LoginException
import play.pluv.login.exception.LoginExceptionType.INVALID_TESTER_ID_PASSWORD

@Component
class TesterLoginValidator(
    @param:Value("\${tester.secret}") private val validSecret: String,
    @param:Value("\${tester.id}") private val testerId: Long
) {
    fun getTesterId(id: String, password: String): Long {
        val validSecret = DigestUtils.sha256Hex(id + password)
        if (this.validSecret != validSecret) {
            throw LoginException(exceptionType = INVALID_TESTER_ID_PASSWORD)
        }
        return testerId
    }
}
